hw3/pos_sentence.py

```diff
@@ -1,9 +1,14 @@
+UNKNOWN = '<UNK>'
+START_WORD = 'START'
+STOP_WORD = 'STOP'
+START_TAG = '^'
+STOP_TAG = '$'
 class POSSentence:
-    def __init__(self, words, tag_sequence=None):
-        self.words = words
-        self.tags = tag_sequence
+    def __init__(self, words, tag_sequence=None, ngram=3):
+        self.words = [START_WORD]*(ngram-1) + words + [STOP_WORD]*(ngram-1)
+        self.tags = [START_TAG]*(ngram-1) + tag_sequence + [STOP_TAG]*(ngram-1)
```

Comment:
we also need to handle the case where tag_sequence is None, i.e. when we read test file

Reply:
And same question as above about multiple end tag, should it be just one tag, start tags are reasonable